gdal/doc/source/development/rfc/rfc78_osgeo_utils.rst

```diff
@@ -0,0 +1,184 @@
+
+* swig/python/osgeo/auxiliary - The `osgeo.auxiliary` subpackage which was introduced very recently
+    in Master as an auxiliary package for `osgeo.utils`.
+    It contains classes and functions that might not be useful by themselves and are used as
+    These shall be moved to `osgeo_utils.auxiliary`.
+
+* swig/python/samples - A bunch of sample scripts for the `osgeo` package.
+    A minimal documentation is available under the README.md file and no tests or backwards compatibility is checked.
+    This RFC suggests that they would be moved into `osgeo_utils.samples` so they can be found easily by users.
+
+* swig/python/scripts - deprecated scripts which are now just a trivial thin wrapper around each of the
+    osgeo.utils modules. These will still be here but will reference `osgeo_utils` instead of `osgeo.utils`.
+
+The layout of `osgeo_utils` would be:
+
+    * `osgeo_utils` root - drop in replacement for `osgeo.utils`.
+    * `osgeo_utils.auxiliary` - drop in replacement for `osgeo.auxiliary`.
+    * `osgeo_utils.samples` - new subpackage for all the samples.
+    * `osgeo_utils.base` - new subpackage for auxiliary modules that do not depend on the `osgeo` package.
+
+
+Folder structure change
+--------------------------
+
+    * `swig/python/osgeo/utils` -> `python/src/osgeo_utils`
+    * `swig/python/osgeo/auxiliary` -> `python/src/osgeo_utils/auxiliary`
+    * `swig/python/scripts` -> `python/src/osgeo_utils/scripts`
+
+
+Backward compatibility
+----------------------
+
+* `osgeo.utils` to be replaced with `osgeo_utils` - The only breaking change breaking change from GDAL 3.2 (One character).
+* `osgeo_utils.*.*` - No backward compatibility is promised for any module inside any subpackage of `osgeo_utils`.
+    So only the modules right inside the root of `osgeo_utils` should keep a steady API.
+* `osgeo_utils.samples` - Maintainers are encouraged to upgrade useful samples
+    into a modules inside the root of `osgeo_utils` by adding more proper documentation, tests and functionality.
+    This will trigger backwards compatibility consideration from that point forward for the upgraded module.
+
+CI Impacts:
+------------
+
+Minor changes to the CI might be required to use the new package and its layout.
+
+Impacts on GDAL core
+--------------------
+
+There should be no impacts on GDAL core.
+
+Limitations and scope
+---------------------
+
+The scope of this RFC should be the GDAL Python code alone, except for the SWIG bindings.
+There shouldn't be effect on any other language supported by GDAL.
+
+SWIG binding changes
+--------------------
+
+No change.
+
+Security implications
+---------------------
+
+None.
+
+Performance impact
+------------------
+
+None.
+
+Documentation
+-------------
+
+The new layout should be documented, maybe inside each sub-package __init__.py file.
+
+Testing
+-------
+
+No any additional tests should be needed.
```

Comment:
If there are 2 separate Python packages, there's probably an impact on the release procedure (HOWTO_RELEASE) regarding update to pypi